### PR TITLE
docs(apim): add gko monitoring section

### PR DESCRIPTION
**Pull request by Okhelifi**
**Issue**

https://gravitee.atlassian.net/jira/software/c/projects/APIM/boards/3?modal=detail&selectedIssue=APIM-361

**Description**

Add a section in the operator about the monitoring using prometheus and grafana 

**Additional context**

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://graviteedocs.blob.core.windows.net/apim-361-gko-monitoring/index.html)

### CODE REVIEW

File: _data/sidebars/apim_3_x_sidebar.yml [hunk 1]
```diff
@@ -703,6 +703,9 @@ entries:
                   - title: Setting up a local cluster for deployment
                     output: web
                     url: /apim/3.x/apim_kubernetes_operator_installation_local.html
+                  - title: Monitor the Gravitee Kubernetes Operator
```
Comment (ivan-gravitee) [inline _data/sidebars/apim_3_x_sidebar.yml:706]:
```suggestion
                  - title: Monitoring GKO
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 2]
```diff
@@ -0,0 +1,98 @@
+[[apim-kubernetes-operator-monitoring]]
+= Monitor the Gravitee Kubernetes Operator
+:page-sidebar: apim_3_x_sidebar
+:page-permalink: apim/3.x/apim-kubernetes-operator-monitoring.html
+:page-folder: apim/kubernetes
+:page-layout: apim3x
+
+[label label-version]#New in version 3.19.0#
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:8]:
```suggestion
[label label-version]#New in version 3.20.0#
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 3]
```diff
@@ -0,0 +1,98 @@
+[[apim-kubernetes-operator-monitoring]]
+= Monitor the Gravitee Kubernetes Operator
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:2]:
```suggestion
= Monitoring the Gravitee Kubernetes Operator
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+:page-layout: apim3x
+
+[label label-version]#New in version 3.19.0#
+[label label-version]#BETA release#
+
+== Overview
+
+This section provides steps on how to monitor the Gravitee Kubernetes Operator (GKO) on an existing APIM-ready Kubernetes cluster.
+
+NOTE: If you do not have an existing Gravitee Kubernetes Operator, you can set up a new one - see the link:{{ '/apim/3.x/apim-kubernetes-operator-installation-cluster.html' | relative_url }} for details. Once you have the operator set up and running, return to this section to deploy the GKO.
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:15]:
```suggestion
NOTE: If you do not have an existing Gravitee Kubernetes Operator deployment, you can easily set up a new one - see link:{{ '/apim/3.x/apim-kubernetes-operator-installation-cluster.html' | relative_url }}[Deploying GKO on a Kubernetes cluster] for details. Once you have the operator set up and running, return to this page to deploy GKO monitoring.
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+....
+
+The command-line output should include `controller-manager-metrics-monitor`, as shown in the example below:
+
+....
+NAME                                 AGE
+controller-manager-metrics-monitor   21s
+....
+
+NOTE: You're now ready to play with the operator. However, most of the actions described here are not persistent and should be mainly used for locally running or dev purposes.
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:98]:
```suggestion
NOTE: Most of the actions described above are not persistent and should mainly be used for development/test purposes and not in production.
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+== Overview
+
+This section provides steps on how to monitor the Gravitee Kubernetes Operator (GKO) on an existing APIM-ready Kubernetes cluster.
+
+NOTE: If you do not have an existing Gravitee Kubernetes Operator, you can set up a new one - see the link:{{ '/apim/3.x/apim-kubernetes-operator-installation-cluster.html' | relative_url }} for details. Once you have the operator set up and running, return to this section to deploy the GKO.
+
+== Prerequisites
+
+Before you start the deployment process, ensure that you have access to the following libraries:
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:19]:
```suggestion
Before you start the monitoring deployment process, ensure that you have access to the following libraries:
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+- The kubectl command-line tool installed on your local machine, configured to connect to your cluster. You can read more about installing kubectl in the official documentation.
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:21]:
```suggestion
- The kubectl command-line tool installed on your local machine, configured to connect to your cluster. Learn more about installing kubectl in the official Kubernetes documentation.
```

Comment (ivan-gravitee) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:21]: Perhaps provide a link?

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+- The Helm package manager installed on your local machine. You can read more about installing Helm in the link:https://helm.sh/docs/intro/install/[official documentation].
+
+== Deploying the Prometheus Stack
+
+The link:https://github.com/prometheus-operator/kube-prometheus[kube-prometheus repository] is a collection of Kubernetes manifests, link:https://grafana.com/[Grafana] dashboards, and link:https://prometheus.io/[Prometheus rules] combined with documentation and scripts to provide easy to operate end-to-end Kubernetes cluster monitoring with Prometheus using the Prometheus Operator.
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:27]:
```suggestion
The link:https://github.com/prometheus-operator/kube-prometheus[kube-prometheus repository] is a collection of Kubernetes manifests, link:https://grafana.com/[Grafana] dashboards, and link:https://prometheus.io/[Prometheus rules], combined with documentation and scripts to provide easy-to-operate, end-to-end Kubernetes cluster monitoring with Prometheus using the Prometheus Operator.
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+=== STEP 1: Run the deployment script
+
+Run the following command in your command-line tool (the working directory does not matter) create the namespace which will be used to deploy the Prometheus stack:
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:31]:
```suggestion
Run the following command in your command-line tool (the working directory does not matter) to create the namespace to be used to deploy the Prometheus stack:
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+....
+kubectl create namespace monitoring
+kubectl config set-context --current --namespace=monitoring
+....
+
+Before installation, please check you’ve added the dedicated repository to helm:
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:38]: This part should be moved above the installation part above

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+....
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+helm install prometheus prometheus-community/kube-prometheus-stack --debug
+....
+
+Now we can check if the Prometheus and Grafana has been installed correctly.
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:46]:
```suggestion
Check if Prometheus and Grafana have been installed correctly:
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+prometheus-operated                       ClusterIP   <none>        9090/TCP                     2m51s
+....
+
+To access the Grafana dashboard, run the following command:
+
+....
+kubectl port-forward -n monitoring svc/prometheus-grafana 8000:80
+....
+
+Grafana is now available at http://localhost:8000. By default an admin user is created. The password can be easily found running the following command:
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:71]:
```suggestion
Grafana is now available at http://localhost:8000. An admin user is created by default. To find out the password for that user, run the following command:
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+
+....
+k get secrets prometheus-grafana -n monitoring -o jsonpath='{.data.admin-password}'|base64 -d
+....
+
+=== STEP 2: Link the Prometheus stack to the GKO
+
+The Prometheus resource declaratively describes the desired state of a Prometheus deployment, while ServiceMonitor resources describe the targets to be monitored by Prometheus. The Gravitee Kubernetes Operator (GKO) provides a basic ServiceMonitor resource that can be discovered by Prometheus. To enable this, you need to run the following command:
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:79]:
```suggestion
The Prometheus resource declaratively describes the desired state of a Prometheus deployment, while `ServiceMonitor` resources describe the targets to be monitored by Prometheus. The Gravitee Kubernetes Operator (GKO) provides a basic `ServiceMonitor` resource that can be discovered by Prometheus. To enable this, run the following command:
```

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+....
+kubectl apply -f https://raw.githubusercontent.com/gravitee-io/gravitee-kubernetes-operator/alpha/config/prometheus/monitor.yaml
+....
+
+To check if the ServiceMonitor has been created, run the following command:
```
Comment (ivan-gravitee) [inline pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:85]:
```suggestion
To check if the `ServiceMonitor` resource has been created, run the following command:
```

**Review by ivan-gravitee — CHANGES_REQUESTED** [review]
Looks good @Okhelifi - I've added a few suggestions for better wording, let me know if any questions:)

File: pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc (hunk 10 above)
Comment (Okhelifi) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc:38]: No I'm not sure sorry 😅. This is really logical to do it here according to the documentation I wrote and what the user will do. 
However I agree maybe we could rephrase this line. I proposed something else in my last push on the branch so feel free to tell me if it's better or if you really think we should move this section.

Comment (ivan-gravitee) [reply pages/apim/3.x/kubernetes/apim-kubernetes-operator-monitoring.adoc]: It's perfectly clear now with the rewritten sub-step, thanks @Okhelifi :)

**Review by ivan-gravitee — APPROVED** [review]
LGTM

**Pull request merged**